### ISSUE TITLE
iOS bundle/build regex might be to restrictive

### ISSUE THREAD

**Issue by pellekrogholt** [SamVerschueren/gulp-cordova-version#2]
```
Config.prototype.setIOSBundleVersion = function(version) {
    var regex = new RegExp('[0-9]+\.[0-9]+\.[0-9]+');

    if(!regex.test(version)) {
        // If the version is not valid, throw an error.
        throw new Error('Please provide a valid version number.');
    }

    // Set the version of the widget tag
    this._root.attrib['ios-CFBundleVersion'] = version;
};
```

lets say I simply provide 

```
.pipe(version(1.0.0, { iosBundleVersion: '1'}))
```

then `gulp-cordova-version` fails with:

```
Error: Please provide a valid version number.
```

That is awkward - first of all because its actually the build version that fails.

iOS build version uses the same regex as iOS version that might really not be suitable - here is a long stackoverflow post on iOS version vs. bundle/build version:
- http://stackoverflow.com/questions/6851660/version-vs-build-in-xcode
- http://stackoverflow.com/questions/6876923/difference-between-xcode-version-cfbundleshortversionstring-and-build-cfbundl

NB: I'm for sure motivated to provide a PR for this issue but would like some feedback to ensure we are on the same page concerning `the solution`.

**Comment by SamVerschueren** [comment]
Hi Pelle, thanks for letting me know. I am not really familiar with iOS development and only had a look at the example of Cordova where they use the `x.y.z` pattern for the bundle version.

You are totally right that the error message is not correct, this should be fixed. Or the check should be removed in general?

What would be your solution on this? Is it possible for instance to have a version number like `abc` or should it be numeric like `1`, `123`, `1.123`, `1.2.3`?

And a PR would be very much appreciated!

**Comment by pellekrogholt** [comment]
@SamVerschueren whats your thoughts about the above ? I'm on a wrong path here ?

**Comment by SamVerschueren** [comment]
Hi, yesterday I thought you weren't :). From the links you provide it seems the CFBundleVersion could be a number like `5123` as well.

But when I read this post:
http://stackoverflow.com/questions/7281085/whats-the-difference-between-version-number-in-itunes-connect-bundle-versio

And I quote

> As pointed out by snlehton, the CFBundleVersion has to be a pure version number like 1.2.3 when you want to upload your application to the (iOS) App Store.

Any thoughts on this? As said before, iOS is not my cup of tea so any information on this is very welcome :).

**Comment by zezke** [comment]
If we take a look at the documentation at https://developer.apple.com/library/ios/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102364 it states:

```
CFBundleVersion (String - iOS, OS X) specifies the build version number of the bundle, which identifies an iteration (released or unreleased) of the bundle. 
The build version number should be a string comprised of three non-negative, period-separated integers with the first integer being greater than zero. The string should only contain numeric (0-9) and period (.) characters. Leading zeros are truncated from each integer and will be ignored (that is, 1.02.3 is equivalent to 1.2.3). This key is not localizable.
```

So seems like the regex is too strict: it could be x, x.y or x.y.z where both x, y and z are non-negative integers and x being at least 1.

**Comment by SamVerschueren** [comment]
@zezke The error you had on your computer said "at most three non-negative integers", hence the words `at most`. The documentation you refering to is not stating `at most` meaning it can only be `x.y.z`...

Damn you apple documentation :)

![pasted image at 2015_09_07 09_17 am](https://cloud.githubusercontent.com/assets/1913805/9719769/291d7fac-5588-11e5-940e-8f8ea1fcc7cc.png)

**Comment by SamVerschueren** [comment]
Resolved in `cordova-config#v0.6.2`

**Comment by WuglyakBolgoink** [comment]
Hallo @SamVerschueren,

in our "pre-release" App we use Version like v0.xx.xx.
with your regex I can't create project ;( 

I read https://github.com/SamVerschueren/gulp-cordova-version/issues/2#issuecomment-138228784, but... Any tip to help me?
